Disambiguate Variable Length Integer

A60AB5450353F40E opened the pull request:
This is used in production code and defined here: https://gitlab.com/bitcoin-cash-node/bitcoin-cash-node/-/blob/master/src/serialize.h#L353

https://github.com/BitcoinUnlimited/BitcoinCashSpecification/issues/32

A60AB5450353F40E commented:
I hurried a bit to make the PR, maybe your idea was to have only consensus stuff in there, and the other VarInt doesn't belong there as it's used for local storage while CompactInt is in the TX raw format so is consensus. I feel it needs to be disambiguated from the other VarInt encoding... simplest could be to rename [this](https://reference.cash/protocol/formats/variable-length-integer) to "Compact Variable Length Integer"

gandrewstone commented:
I'm fine with adding nonconsensus stuff.  All the network protocols are non-consensus.  Why did you close this PR?

A60AB5450353F40E committed:
Disambiguation: rename variable length integer to **compact** variable length integer

See issue [#32](https://github.com/BitcoinUnlimited/BitcoinCashSpecification/issues/32)

Quoting @joshmg:

"I think we're inclined to take your most recent suggestion and rename what the specification calls "variable length integer" to "compact variable length integer". It's similar enough to the C++ -ism such that anyone with that context will understand the intended parallel, while also avoiding the name collision of BCHN/Core's internal VarInt."

A60AB5450353F40E commented:
"Why did you close this PR?" because I went ahead of myself and then quickly realized it's not the way to address the confusion. I updated it just now after some discussion here: https://github.com/BitcoinUnlimited/BitcoinCashSpecification/issues/32

joshmg commented:
I spoke with @thesquaregroot and we're going to accept and merge this PR and then expand on it a (very) little bit to further disambiguate the difference between the spec and the C++ codebases, but we're planning on doing this within the Software Verde repo which will then get propagated here.
